packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1136 @@
+      _kDrag,
+      translation.dx,
+      details.velocity.pixelsPerSecond.dx,
+    );
+    final FrictionSimulation frictionSimulationY = FrictionSimulation(
+      _kDrag,
+      translation.dy,
+      details.velocity.pixelsPerSecond.dy,
+    );
+    final double tFinal = _getFinalTime(
+      details.velocity.pixelsPerSecond.distance,
+      _kDrag,
+    );
+    _animation = Tween<Offset>(
+      begin: translation,
+      end: Offset(frictionSimulationX.finalX, frictionSimulationY.finalX),
+    ).animate(CurvedAnimation(
+      parent: _controller,
+      curve: Curves.decelerate,
+    ));
+    _controller.duration = Duration(milliseconds: (tFinal * 1000).round());
+    _animation.addListener(_onAnimate);
+    _controller.forward();
+  }
+
+  // Handle mousewheel scroll events.
+  void _receivedPointerSignal(PointerSignalEvent event) {
+    if (event is PointerScrollEvent) {
+      final Size childSize = _getChildSize(
+        _childKey.currentContext.findRenderObject() as RenderBox,
+        _constraints,
+      );
+      final double scaleChange = 1.0 + event.scrollDelta.dy / childSize.height;
+      if (scaleChange == 0.0) {
+        return;
+      }
+      final Offset focalPointScene = widget.transformationController.toScene(
+        event.localPosition,
+      );
+      setState(() {
+        widget.transformationController.value = _matrixScale(
+          widget.transformationController.value,
+          scaleChange,
+        );
+
+        // After scaling, translate such that the event's position is at the
+        // same scene point before and after the scale.
+        final Offset focalPointSceneScaled = widget.transformationController.toScene(
+          event.localPosition,
+        );
+        widget.transformationController.value = _matrixTranslate(
+          widget.transformationController.value,
+          focalPointSceneScaled - focalPointScene,
+        );
+      });
+    }
+  }
+
+  // Handle inertia drag animation.
+  void _onAnimate() {
+    if (!_controller.isAnimating) {
+      _animation?.removeListener(_onAnimate);
+      _animation = null;
+      _controller.reset();
+      return;
+    }
+    // Translate such that the resulting translation is _animation.value.
+    final Vector3 translationVector = widget.transformationController.value.getTranslation();
+    final Offset translation = Offset(translationVector.x, translationVector.y);
+    final Offset translationScene = widget.transformationController.toScene(
+      translation,
+    );
+    final Offset animationScene = widget.transformationController.toScene(
+      _animation.value,
+    );
+    final Offset translationChangeScene = animationScene - translationScene;
+    setState(() {
+      widget.transformationController.value = _matrixTranslate(
+        widget.transformationController.value,
+        translationChangeScene,
+      );
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
```

Comment:
Maybe I missed this, but I would have expected that you'd listen to the `TransformationController` somewhere to rebuild when the transformation changes?

Reply:
Done as part of https://github.com/flutter/flutter/pull/56409#discussion_r431324470